src/main/java/com/wavefront/sdk/direct/ingestion/WavefrontDirectIngestionClient.java

```diff
@@ -405,4 +415,54 @@ public synchronized void close() {
       logger.log(Level.WARNING, "shutdown error", ex);
     }
   }
+
+  /**
+   * Dequeue and return a batch of at most N items from buffer (where N = batchSize), broken into
+   * chunks where each chunk has at most M bytes of data (where M = messageSizeBytes).
+   *
+   * Visible for testing.
```

Comment:
nit: if you have annotation `@VisibleForTesting` already in one of the dependencies would prefer that.

Reply:
it's not in any of the current dependencies -- didn't want to add google dependencies to make this happen